contracts/core/src/solve/Solve.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity =0.8.24;
+
+library Solve {
+    /**
+     * @notice A request to execute a call on another chain, backed by a deposit.
+     * @param id            ID for the request, globally unique per inbox.
+     * @param openedAt      Timestamp of when the request was created.
+     * @param from          Address of the user who created the request.
+     * @param accepted      Whether the request has been accepted.
+     * @param rejected      Whether the request has been rejected.
+     * @param cancelled     Whether the request has been canceled.
+     * @param fulfilled     Whether the request has been fulfilled.
+     * @param paid          Whether the request has been paid.
+     * @param fulfilledBy   Address of the entity that fulfilled the request.
+     * @param call          Details of the call to be executed on another chain.
+     * @param deposits      Array of deposits backing the request.
+     */
+    struct Request {
+        uint256 id;
+        uint96 openedAt;
+        address from;
+        bool accepted;
+        bool rejected;
+        bool cancelled;
+        bool fulfilled;
+        bool paid;
```

Comment:
Going to switch these to a uint8 status. doesn't change number of slots packed

Reply:
Could use an enum too, it's a uint8 under the hood